src/main/scala/beam/router/RoutingStatistic.scala

```diff
@@ -21,8 +22,9 @@ class RoutingStatistic(ioController: OutputDirectoryHierarchy) extends Actor {
 
   def handleRouterResponses(writer: CsvWriter): Receive = {
     case _: IterationEndsMessage =>
-      writer.flush()
       writer.close()
+      context.become(receive)
```

Comment:
this will initialize a new writer, I wonder if this will overwrite the existing data in that file from the previous iteration?

Reply:
This would only be the case if it is the same iteration. If we want appending then we would have to modify the CsvWriter  to have the ability to call `getAppendingBufferedWriter`.

Reply:
When we call `context.become(receive)`, the previous writer is closed. And we open a new one with a new file when `IterationStartsMessage` arrives.

Reply:
Yah, that should be guarantee enough :)